internal/pkg/deploy/pipeline.go

```diff
@@ -82,78 +85,84 @@ type Source struct {
 // Secrets manager, which stores the GitHub Personal Access token if the
 // provider is "GitHub". Otherwise, it returns the detected provider.
 func (s *Source) GitHubPersonalAccessTokenSecretID() (string, error) {
-	// TODO type check if properties are GitHubProperties?
-	secretID, exists := s.Properties[manifest.GithubSecretIdKeyName]
-	if !exists {
-		return "", errors.New("the GitHub token secretID is not configured")
-	}
-
-	id, ok := secretID.(string)
-	if !ok {
-		return "", fmt.Errorf("unable to locate the GitHub token secretID from %v", secretID)
-	}
+	id := "N/A"
```

Comment:
Why do we need `N/A` instead of just an empty string?

Reply:
Hmm interesting, is it possible to have two different types: `GitHubSource` and `CodeCommitSource`? where this method is declared only for a `*GitHubSource`.

I really like this lesson from Effective Java: [Prefer Class Hierarchies to Tagged Classes](https://dev.to/kylec32/effective-java-prefer-class-hierarchies-to-tagged-classes-59l9). Since we have a few methods of the following pattern, maybe there is an opportunity for us to refactor?
```go
if s.ProviderName == manifest.GithubProviderName {
  ...
}
if s.ProviderName == manifest.CodeCommitProviderName {
  ...
}
```

So the idea is to have separate methods like this:
```go
func (s *GitHubSource) PersonalAccessTokenSecretID() (string, error)
func (s *GitHubSource) Owner() (string, error)
func (s *GitHubSource) Repo() (string, error)

func (s *CodeCommitSource) Repo() (string, error)
```
Then, the input to the stack template can be this maybe:
```go
type CreatePipelineInput struct {
    Source interface{}
}
```

Reply:
This is the error I got when I tried to leave it as an empty string:
`✘ update pipeline: update pipeline: check if changeset is empty: create change set copilot-94adeb7f-bb8a-417e-b3e4-04a7be96e94c for stack pipeline-sample-aws-sample: ValidationError: [/Resources/Pipeline/Type/Stages/0/Actions/Fn::If/1/0/Configuration/OAuthToken/Fn::Sub/1/SecretId] 'null' values are not allowed in templates`
I'd prefer to leave this (and Owner) as empty strings, though; I'll think on this!

Reply:
> Hmm interesting, is it possible to have two different types: `GitHubSource` and `CodeCommitSource`? where this method is declared only for a `*GitHubSource`.
> 

Ah, gotcha-- this is like what we did with the parsing methods in the last PR. 👍🏼